### PR TITLE
Release 2.14.0

### PR DESCRIPTION
**Pull request by Aleffio**

### CODE REVIEW

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 1]
```diff
@@ -66,6 +66,353 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+            if (!empty($recurringType) && $recurringType !== self::RECURRING_TYPE_ONECLICK) {
+                $request['enableRecurring'] = true;
+            }
+        } elseif ($paymentMethod === 'cc') {
+            $request['enableRecurring'] = false;
+            $request['enableOneClick'] = true;
+
+            // if save card is disabled only shoot in as recurring if recurringType is set to ONECLICK,RECURRING
+            if ($payment->getAdditionalInformation('store_cc') == '' &&
+                $recurringType === 'ONECLICK,RECURRING'
```
Comment (cyattilakiss) [inline app/code/community/Adyen/Payment/Model/Api.php]: use the constants in this file, like: RECURRING_TYPE_ONECLICK_RECURRING

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 2]
```diff
@@ -66,6 +66,353 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+                $request['paymentMethod']['type'] = 'scheme';
+                if (!empty($payment->getCcOwner())) {
+                    $request['paymentMethod']['holderName'] = $payment->getCcOwner();
+                }
+
+                if (!empty($encryptedNumber) && $encryptedNumber != "false") {
+                    $request['paymentMethod']['encryptedCardNumber'] = $encryptedNumber;
+                }
+
+                if (!empty($encryptedExpiryMonth) && !empty($encryptedExpiryYear != "")) {
```
Comment (cyattilakiss) [inline app/code/community/Adyen/Payment/Model/Api.php]:
```suggestion
                if (!empty($encryptedExpiryMonth) && !empty($encryptedExpiryYear)) {
```

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 3]
```diff
@@ -66,6 +66,353 @@ public function getRecurringContractDetail($shopperReference, $recurringDetailRe
...
+     * Create a 3D secure payment request
+     *
+     * @param $payment
+     * @return mixed
+     */
+    public function authorise3DPayment($payment)
+    {
+        $apiKey = $this->_helper()->getConfigDataApiKey(null);
+        if ($this->_helper()->getConfigDataDemoMode()) {
+            $requestUrl = "https://checkout-test.adyen.com/v41/payments/details";
```
Comment (cyattilakiss) [inline app/code/community/Adyen/Payment/Model/Api.php]: you can build it the same way as the live one

File: app/code/community/Adyen/Payment/Model/Api.php (hunk 1 above)
Comment (Aleffio) [reply app/code/community/Adyen/Payment/Model/Api.php]: done

File: app/code/community/Adyen/Payment/Model/Api.php (hunk 3 above)
Comment (Aleffio) [reply app/code/community/Adyen/Payment/Model/Api.php]: done

**Pull request merged**